src/request.rs

```diff
@@ -194,71 +220,94 @@ impl RequestBuilder {
     }
 
     /// Enable HTTP bearer authentication.
-    pub fn bearer_auth(self, token: impl Into<String>) -> RequestBuilder {
+    pub fn bearer_auth(self, token: impl Into<String>) -> Self {
         self.header(http::header::AUTHORIZATION, format!("Bearer {}", token.into()))
     }
 
+    /// Set the body of this request.
+    ///
+    /// The can be a `&[u8]` or a `str`, anything that's a sequence of bytes.
+    pub fn body(self, body: impl AsRef<[u8]>) -> RequestBuilder<impl AsRef<[u8]>> {
+        RequestBuilder {
+            url: self.url,
+            method: self.method,
+            headers: self.headers,
+            body,
+            max_redirections: self.max_redirections,
+            follow_redirects: self.follow_redirects,
+            #[cfg(feature = "charsets")]
+            default_charset: self.default_charset,
+            #[cfg(feature = "compress")]
+            allow_compression: self.allow_compression,
+        }
+    }
+
     /// Set the body of this request to be text.
     ///
     /// If the `Content-Type` header is unset, it will be set to `text/plain` and the carset to UTF-8.
-    pub fn text(mut self, body: impl Into<String>) -> RequestBuilder {
-        self.body = body.into().into_bytes();
+    pub fn text(mut self, body: impl AsRef<str>) -> RequestBuilder<impl AsRef<[u8]>> {
+        struct Text<B1>(B1);
+
+        impl<B1: AsRef<str>> AsRef<[u8]> for Text<B1> {
+            fn as_ref(&self) -> &[u8] {
+                self.0.as_ref().as_bytes()
+            }
+        }
+
         self.headers
             .entry(http::header::CONTENT_TYPE)
             .unwrap()
             .or_insert(HeaderValue::from_static("text/plain; charset=utf-8"));
-        self
+        self.body(Text(body))
```

Comment:
Doesn'tt `body.as_ref()` work here? `str` implements `AsRef<u8>`

Reply:
I checked out the code and I see now that you need to own the `impl AsRef<str>` where `.as_ref()` simply borrows it and that doesn't work. Neat work around.